### PR TITLE
ref(tsdb): Remove TSDB increments from track_outcome

### PR DESCRIPTION
**Pull request by untitaker**
1. Relay will be required for store
2. Relay cannot increment redis TSDB directly
3. We used to have a TSDB-incrementing outcomes consumer in sentry. That consumer is gone since we moved it into getsentry, where it does no longer do that.
4. Therefore the only way to get all metrics to work right now is to use the redissnuba backend (or the almost identical service delegator in getsentry as discovered in #19760).
5. If redissnuba is already mandatory to get anything working, this is entirely useless as the snuba outcomes consumer will take care of capturing this data from outcomes.

**Verify:** Snuba outcomes consumer is running in

- [x] singletenant
- [x] onpremise
- [x] saas (that's how #19760 started)

### CODE REVIEW

**Comment by untitaker** [comment]
Updated development docs here: https://develop.sentry.dev/services/tsdb/

File: tests/sentry/event_manager/test_event_manager.py [hunk 1]
```diff
@@ -1049,19 +1049,6 @@ def test_throws_when_matches_discarded_hash(self):
...
-
-        # Ensure that we incremented TSDB counts
-        assert query(tsdb.models.organization_total_received, event.project.organization.id) == 2
-        assert query(tsdb.models.project_total_received, event.project.id) == 2
-
-        assert query(tsdb.models.project, event.project.id) == 1
-        assert query(tsdb.models.group, event.group.id) == 1
-
-        assert query(tsdb.models.organization_total_blacklisted, event.project.organization.id) == 1
-        assert query(tsdb.models.project_total_blacklisted, event.project.id) == 1
```
Comment (jan-auer) [inline tests/sentry/event_manager/test_event_manager.py:1063]: This test is actually pretty important for org stats. Are we sure that this is properly tested for the other backend?

**Review by jan-auer — COMMENTED** [review]
Partial review. Let's verify that Snuba is now the only supported TSDB backend for outcomes related models.

File: tests/sentry/event_manager/test_event_manager.py (hunk 1 above)
Comment (untitaker) [reply tests/sentry/event_manager/test_event_manager.py:1063]: The snuba backend is quite well-tested: https://github.com/getsentry/sentry/blob/ref/sentry-inbound-filters/tests/sentry/tsdb/test_snuba.py

As long as we make sure `track_outcome` is called in EventManager, and make sure outcomes result in tsdb incrs (which we do in the snuba tsdb testsuite), I think we should be good.

Comment (jan-auer) [reply tests/sentry/event_manager/test_event_manager.py:1063]: Alright, considering that outcomes are checked just above this, let's go ahead with this 👍

**Review by jan-auer — APPROVED** [review]

**Comment by fpacifici** [comment]
Sorry just one clarification around your plan.
Do you plan to adapt redissnuba so that it behaves like service delegator does in prod? Or do you plan to make on prem work based on service delegator? 
It seems to me we should update everything to redissnuba now that the consumer is there everywhere and make it actually load outcomes from the right place in all cases.

**Comment by untitaker** [comment]
>Do you plan to adapt redissnuba so that it behaves like service delegator does in prod? Or do you plan to make on prem work based on service delegator?

I don't see the difference in their behavior right now, so I would like to adopt redissnuba everywhere. That was kind of the plan from the start as the rate-limited outcomes cannot be obtained from redis. redissnuba reads outcomes-based tsdb metrics since https://github.com/getsentry/sentry/pull/19760

File: src/sentry/utils/outcomes.py [hunk 2]
```diff
@@ -135,26 +64,6 @@ def track_outcome(
 
     timestamp = timestamp or to_datetime(time.time())
```
Comment (fpacifici) [inline src/sentry/utils/outcomes.py:66]: Please update the docstring of the method that still says we are updating redis.

File: src/sentry/utils/outcomes.py [hunk 3]
```diff
@@ -135,26 +64,6 @@ def track_outcome(
...
-                reason=reason,
-                category=category,
-            )
-        )
-
-        if increment_list:
-            tsdb.incr_multi(increment_list, timestamp=timestamp)
-
-        if project_id and event_id:
-            mark_tsdb_incremented(project_id, event_id)
```
Comment (fpacifici) [inline src/sentry/utils/outcomes.py:156]: Did you verify none, except for the consumer that is supposed to, actually depends on this cache ?

**Review by fpacifici — COMMENTED** [review]
Please see my question inline

File: src/sentry/utils/outcomes.py (hunk 3 above)
Comment (untitaker) [reply src/sentry/utils/outcomes.py:156]: yes

**Comment by untitaker** [comment]
checking with ops that we really don't use redis tsdb in prod for this stuff anymore

**Comment by fpacifici** [comment]
Ok, then it is ok to me (as long as you verify this is not used anymore outside of the codebase).

**Review by fpacifici — APPROVED** [review]

**Pull request merged**